docs/dev/setup/jenkins-gitlab.rst

```diff
@@ -733,6 +733,95 @@ Configure your master node like this  (adjust the number of executors, if needed
 
       Jenkins local node
 
+Alternative local build agents setup using docker
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An alternative way of adding a build agent that will use docker (similar to the remote agents below) but running
+locally, can be done using the jenkins/ssh-agent docker image (https://hub.docker.com/r/jenkins/ssh-agent).
+
+Prerequisites:
+
+1. Make sure to have Docker installed: https://docs.docker.com/engine/install/ubuntu/
+
+Agent setup:
+
+1. Create a new SSH key using ``ssh-keygen`` (if a passphrase is added, store it for later)
+
+2. Copy the public key content (e.g. in ~/.ssh/id_rsa.pub)
+
+3. Run ``docker run -d --name jenkins_agent -v /var/run/docker.sock:/var/run/docker.sock jenkins/ssh-agent:latest "<copied_public_key>"``
+
+4. Get the GID of the 'docker' group with ``cat /etc/groups`` and remember it for later
+
+5. Enter the agent's container with ``docker exec -it jenkins_agent bash``
+
+6. Install Docker with ``apt update && apt install docker.io``
+
+7. Check if group 'docker' already exists with ``cat /etc/groups``. If yes, remove it with ``groupdel docker``
+
+8. Add a new 'docker' group with the same GID as seen in point 2 with ``groupadd -g <GID> docker``
+
+9. Add 'jenkins' user to the group with ``usermod -aG docker jenkins``
+
+10. Activate changes with ``newgrp docker``
+
+11. Now check if 'jenkins' has the needed permissions to run docker commands
+
+    1. Log in as 'jenkins' with ``su jenkins``
+
+    2. Try if ``docker inspect <agent_container_name>`` works or if a permission error occurs
+
+    3. If an permission error occurs, try to restart the docker container
+
+12. Now you can exit the container executing ``exit`` twice (the first will exit the jenkins user and the second the container)
+
+Add agent in Jenkins:
+
+1. Open Jenkins in your browser (e.g. localhost:8082)
+
+2. Go to Manage Jenkins -> Manage Credentials -> (global) -> Add Credentials
+
+    Kind: SSH Username with private key
+
+    ID: leave blank
+
+    Description: Up to you
+
+    Username: jenkins
+
+    Private Key: <content of the previous generated private key> (e.g /root/.ssh/id_rsa)
+
+    Passphrase: <the previous entered passphrase> (you can leave it blank if none has been specified)
```

Comment:
This list as is looks odd to me:
![grafik](https://user-images.githubusercontent.com/72132281/104104778-71143600-52aa-11eb-95a3-a56d7ca48fd7.png)
Could we have a screenshot instead? (With the example values filled in.) Or perhaps a bullet point list/"code block"?